Sources/Commands/SwiftPackageTool.swift

```diff
@@ -552,23 +552,35 @@ extension SwiftPackageTool {
         var includeSPISymbols = false
 
         func run(_ swiftTool: SwiftTool) throws {
-            let symbolGraphExtract = try SymbolGraphExtract(
-                tool: swiftTool.getToolchain().getSymbolGraphExtract())
-
             // Build the current package.
             //
             // We turn build manifest caching off because we need the build plan.
             let buildOp = try swiftTool.createBuildOperation(cacheBuildManifest: false)
             try buildOp.build()
 
-            try symbolGraphExtract.dumpSymbolGraph(
-                buildPlan: buildOp.buildPlan!,
-                prettyPrint: prettyPrint,
-                skipSynthesisedMembers: skipSynthesizedMembers,
+            // Configure the symbol graph extractor.
+            let symbolGraphExtractor = try SymbolGraphExtract(
+                tool: swiftTool.getToolchain().getSymbolGraphExtract(),
+                skipSynthesizedMembers: skipSynthesizedMembers,
                 minimumAccessLevel: minimumAccessLevel,
                 skipInheritedDocs: skipInheritedDocs,
-                includeSPISymbols: includeSPISymbols
-            )
+                includeSPISymbols: includeSPISymbols,
+                prettyPrintOutputJSON: prettyPrint)
```

Comment:
should this be an enumeration instead? so output: .json(pretty: true)

Reply:
Perhaps — what would the other enum cases be?  Or would it just be JSON for now?

It does sound cleaner... I tried to keep this diff bounded by just moving these properties from the function args to the struct properties to make it easier to configure, but this seems reasonable to change.

Reply:
I really like this actually — it makes conceptual sense even if there is just one enum case for now.